MegaQuests/src/main/java/com/elmer/megaquests/MegaQuests.java

```diff
@@ -1,59 +1,50 @@
 package com.elmer.megaquests;
 
+import com.earth2me.essentials.Essentials;
 import com.elmer.megaquests.commands.QuestGUICommand;
+import com.elmer.megaquests.commands.ResetCooldownCommand;
 import com.elmer.megaquests.listeners.JoinQuitListener;
+import com.elmer.megaquests.listeners.QuestGUIListener;
 import com.elmer.megaquests.listeners.questlisteners.BlockBreakQuestsListener;
 import com.elmer.megaquests.listeners.questlisteners.EntityKillQuestsListener;
-import com.elmer.megaquests.listeners.QuestGUIListener;
 import com.elmer.megaquests.managers.CooldownManager;
 import com.elmer.megaquests.managers.QuestManager;
-import com.elmer.megaquests.commands.ResetCooldownCommand;
 import org.bukkit.Bukkit;
-import org.bukkit.configuration.file.FileConfiguration;
-import org.bukkit.configuration.file.YamlConfiguration;
-import org.bukkit.entity.Player;
 import org.bukkit.plugin.java.JavaPlugin;
-import com.earth2me.essentials.Essentials;
 
 import java.io.File;
-import java.io.IOException;
-import java.util.Map;
-import java.util.UUID;
 
 public final class MegaQuests extends JavaPlugin {
 
     // GUI att välja quest (command för det)
     // Quests: Minea diamonds (Blockbreak lsitener), döda mobs (entitydeathlistener), fiska fiskar (fishing listener?)
     // Essentials API för rewards
 
+    private final File dataFile = new File(getDataFolder(), "cooldowns.dat");
     private Essentials essentials;
     private CooldownManager cooldownManager;
     private QuestGUICommand questGUICommand;
-    private EntityKillQuestsListener entityKillQuestsListener;
     private QuestManager questManager;
-    private BlockBreakQuestsListener blockBreakQuestsListener;
-    private ResetCooldownCommand resetCooldownCommand;
-    private File dataFile = new File(getDataFolder(), "cooldowns.dat");
-    private int questAmount;
 
     @Override
     public void onEnable() {
 
         essentials = Essentials.getPlugin(Essentials.class);
         cooldownManager = new CooldownManager(dataFile, this);
-        questGUICommand = new QuestGUICommand(this);
-        entityKillQuestsListener = new EntityKillQuestsListener(this);
+        questGUICommand = new QuestGUICommand(this, questManager);
         questManager = new QuestManager(this);
-        blockBreakQuestsListener = new BlockBreakQuestsListener(this);
+
+        new EntityKillQuestsListener(this);
+        new BlockBreakQuestsListener(this);
```

Comment:
There is no point in assigning variables to a new instance if those variables are not used later. It is easier to make an instance without assigning a variable